gneiss/regression/_ols.py

```diff
@@ -288,36 +314,34 @@ def kfold(self, num_folds=10, **kwargs):
                Prediction mean sum of squares error for each iteration of
                the cross validation.
         """
-        nobs = self.balances.shape[0]  # number of observations (i.e. samples)
+        # number of observations (i.e. samples)
+        nobs = self.response_matrix.shape[0]
         s = nobs // num_folds
         folds = [np.arange(i*s, ((i*s)+s) % nobs) for i in range(num_folds)]
         results = pd.DataFrame(index=['fold_%d' % i for i in range(num_folds)],
                                columns=['model_mse', 'Rsquared', 'pred_mse'],
                                dtype=np.float64)
-        endog = self.balances
-        exog_names = self.results[0].model.exog_names
-        exog = pd.DataFrame(self.results[0].model.exog,
-                            index=self.balances.index,
-                            columns=exog_names)
+
         for k in range(num_folds):
-            train = folds[k]
-            test = np.hstack(folds[:k] + folds[k+1:])
+            test = folds[k]
+            train = np.hstack(folds[:k] + folds[k+1:])
 
-            model_i = _fit_ols(y=endog.iloc[train], x=exog.iloc[train],
-                               **kwargs)
-            res_i = OLSModel(model_i, balances=endog.iloc[train])
-            res_i.fit()
+            res_i = OLSModel(self.response_matrix.iloc[train],
+                               self.design_matrix.iloc[train])
+            res_i.fit(**kwargs)
 
             # model error
-            predicted = res_i.predict()
-            model_resid = ((predicted - endog.iloc[train])**2)
+            p = res_i.predict(X=self.design_matrix.iloc[train]).values
+            r = self.response_matrix.iloc[train].values
+            model_resid = ((p - r)**2)
             model_mse = np.mean(model_resid.sum(axis=0))
             results.loc['fold_%d' % k, 'model_mse'] = model_mse
             results.loc['fold_%d' % k, 'Rsquared'] = res_i.r2
 
             # prediction error
-            predicted = res_i.predict(exog.iloc[test])
-            pred_resid = ((predicted - endog.iloc[test])**2)
+            p = res_i.predict(X=self.design_matrix.iloc[test]).values
+            r = self.response_matrix.iloc[test].values
+            pred_resid = ((p - r)**2)
```

Comment:
These lines I'm not completely sure about.  I'm trying to calculate mse for both the model error and the prediction error.  The degrees of freedom may be a little off - @serenejiang do you have thoughts here wrt `pred_mse` and `model_mse`?